tests/unit/expiry/HgvTrlMostRecentExpiryStrategy.unitTest.ts

```diff
@@ -0,0 +1,147 @@
+import { cloneDeep } from "lodash";
+import testResults from "../../resources/test-results.json";
+import { TestTypeForExpiry } from "../../../src/models/TestTypeforExpiry";
+import { DateProvider } from "../../../src/handlers/expiry/providers/DateProvider";
+import { TestType } from "../../../src/models/ITestResult";
+import { VEHICLE_TYPE } from "../../../src/assets/Enums";
+import { HgvTrlMostRecentExpiryStrategy } from "../../../src/handlers/expiry/strategies/HgvTrlMostRecentExpiryStrategy";
+
+describe("HgvTrlMostRecentExpiryStrategy", () => {
+  let testResultsMockDB: any;
+  let hgvTrlMostRecentExpiryStrategy: HgvTrlMostRecentExpiryStrategy;
+
+  beforeEach(() => {
+    testResultsMockDB = cloneDeep(testResults);
+  });
+
+  afterEach(() => {
+    // reset date to current date
+    hgvTrlMostRecentExpiryStrategy.dateProvider.setTestDate(new Date());
+  });
+
+  context("for hgv vehicle type", () => {
+    describe("with missing expiry date", () => {
+      it("should set the Expiry last day of month + 1 year of testDate", () => {
+        const hgvTestResult = cloneDeep(testResultsMockDB[4]);
+        hgvTestResult.testTypes.forEach((type: TestType) => {
+          type.testTypeId = "94";
+        });
+        const testType = hgvTestResult.testTypes[0];
+        const testTypeForExpiry: TestTypeForExpiry = {
+          testType,
+          vehicleType:
+            VEHICLE_TYPE[
+              hgvTestResult.vehicleType.toUpperCase() as keyof typeof VEHICLE_TYPE
+            ],
+          hasHistory: false,
+          hasRegistration: false,
+          recentExpiry: DateProvider.getEpoc(),
+          regnDate: hgvTestResult.regnDate
+        };
+        hgvTrlMostRecentExpiryStrategy = new HgvTrlMostRecentExpiryStrategy(
+          testTypeForExpiry,
+          new DateProvider()
+        );
+        hgvTrlMostRecentExpiryStrategy.dateProvider.setTestDate(
+          new Date("2020-06-30")
+        );
+        const expectedExpiryDate = new Date("2021-06-30").toISOString();
+        const actualExpiryDate = hgvTrlMostRecentExpiryStrategy.getExpiryDate();
+        expect.assertions(1);
+        expect(actualExpiryDate).toEqual(expectedExpiryDate);
+      });
+    });
+    describe("with valid expiry date", () => {
+      describe("which is exactly 2 months of test date", () => {
```

Comment:
Again, we should be testing more than a single value for the condition, and for the calculation.  We should be checking the two-month window on boundaries of months, including leap years. and the calculation should check that we add a year to a leap year correctly.